### PR TITLE
feat: [#1471] visually identify required tasks on roadmap and task screen

### PR DESCRIPTION
**Pull request by namanaman**

### CODE REVIEW

File: web/public/mgmt/config.yml [hunk 1]
```diff
@@ -213,59 +214,68 @@ collections:
       - label: "Type"
         name: "type"
         widget: "select"
-        options: [ "FUNDING", "CERTIFICATION" ]
```
[the remark this answers is not in the document]
Comment (aloverso) [reply web/public/mgmt/config.yml:216]: is your prettier configured differently? strange that this wouldn't have been autoformatted already

File: web/src/components/TaskHeader.tsx [hunk 2]
```diff
@@ -14,12 +15,28 @@ import {
...
 import React, { ReactElement, useState } from "react";
 import { CongratulatoryDialog } from "./CongratulatoryDialog";
+import { Tag } from "./njwds-extended/Tag";
 
 interface Props {
   task: Task;
   tooltipText?: string;
 }
 
+const RequiredTag = (): ReactElement => (
```
Comment (aloverso) [inline web/src/components/TaskHeader.tsx:25]: if this is going to be its own independent const, I think it should probably live in its own file

File: web/src/display-defaults/tasks/TaskDefaults.ts [hunk 3]
```diff
@@ -8,4 +8,6 @@ export const TaskDefaults: Record<string, string> = {
   unlockedBySingular: "Before starting this task, we recommend that you take a look at:",
   unlockedByPlural: "Before starting this task, we recommend you take a look at:",
   loadingTaskDependencies: "Loading task dependencies…",
+  requiredTagText: "Required",
+  requiredIconAltText: "Exclamation mark",
```
Comment (aloverso) [inline web/src/display-defaults/tasks/TaskDefaults.ts:12]: Do we want/need alt text for this icon? My understanding is that alt text should convey meaning, not a description of what's on the page. So if the exclamation icon is only there to provide a visual cue, I feel like having a screenreader read out "exclamation mark" would be an unpleasant experience.

File: web/src/lib/utils/markdownReader.ts [hunk 4]
```diff
@@ -62,4 +63,5 @@ export type TaskWithoutLinks = {
   callToActionText: string;
   postOnboardingQuestion: string;
   contentMd: string;
+  required?: boolean;
```
Comment (aloverso) [inline web/src/lib/utils/markdownReader.ts:66]: do opportunities have this required tag also? I thought this was just for tasks

File: web/test/pages/[taskUrlSlug].test.tsx [hunk 5]
```diff
@@ -204,6 +204,20 @@ describe("task page", () => {
...
   });
 
+  it("displays required tag in header if task is required", () => {
+    const subject = renderPage(generateTask({ required: true }));
+
+    expect(subject.getByAltText(TaskDefaults.requiredIconAltText)).toBeInTheDocument();
+    expect(subject.getByText(TaskDefaults.requiredTagText)).toBeInTheDocument();
+  });
+
+  it("does not displays required tag in header if task is not required", () => {
```
Comment (aloverso) [inline web/test/pages/[taskUrlSlug].test.tsx:214]: grammar - does not display

File: web/test/pages/roadmap.test.tsx [hunk 6]
```diff
@@ -409,4 +410,46 @@ describe("roadmap page", () => {
...
+    const sectionPlan = subject.getByTestId("section-plan");
+    const allTasks = within(sectionPlan).getAllByRole("listitem");
+
+    expect(allTasks.length).toEqual(2);
+
+    expect(within(allTasks[0]).getByAltText(TaskDefaults.requiredIconAltText)).toBeVisible();
+    expect(within(allTasks[1]).queryByAltText(TaskDefaults.requiredIconAltText)).not.toBeVisible();
+  });
+
+  it("for a required task, hovering over icon renders tooltip", async () => {
```
Comment (aloverso) [inline web/test/pages/roadmap.test.tsx:434]: we like to keep the test names reading as sentences including the "it" (the reason is because it helps ensure that a test is describing the behavior of the component under test). So if we could rephrase this to read like a sentence where the component is the subject, that would be good. For example "it renders tooltip text when..."

File: web/src/lib/utils/helpers.tsx [hunk 7]
```diff
@@ -90,7 +90,7 @@ export const setPreferencesCloseSection = (preferences: Preferences, current: Se
 export const getModifiedTaskContent = (
   roadmap: Roadmap | undefined,
   task: Task,
-  field: Exclude<keyof Task, "unlockedBy">
+  field: Exclude<keyof Task, "unlockedBy" | "required">
```
[the remark this answers is not in the document]
Comment (namanaman) [reply web/src/lib/utils/helpers.tsx:93]: Ah, since it only returned `string`, I was thinking the helper is used to get displayable content like the name and CTA text. Not sure if returning `string | boolean` will fit into the helper's design here, as all the callers are expecting a `string` back to feed into components like `<Content />` or `<TaskCTA />`.

File: web/src/lib/utils/markdownReader.ts (hunk 4 above)
Comment (namanaman) [reply web/src/lib/utils/markdownReader.ts:66]: Just for tasks I think, is the `TaskWithoutLinks` type here also being used for opportunities?

File: web/test/pages/roadmap.test.tsx (hunk 6 above)
Comment (namanaman) [reply web/test/pages/roadmap.test.tsx:434]: Oooh that makes sense

File: web/src/display-defaults/tasks/TaskDefaults.ts (hunk 3 above)
Comment (namanaman) [reply web/src/display-defaults/tasks/TaskDefaults.ts:12]: I guess since there is a tooltip with a label that would get read by screen readers, there's no need for an alt text here then? I imagine a more semantic alt text would end up being the same as the tooltip label.

File: web/public/mgmt/config.yml (hunk 1 above)
[the remark this answers is not in the document]
Comment (namanaman) [reply web/public/mgmt/config.yml:216]: It's the default, perhaps just from VSCode? Changes are changing single quotes to double, indenting long single-line arrays, adding trailing commas, and adding extra spaces around array brackets

File: web/src/components/TaskHeader.tsx (hunk 2 above)
Comment (namanaman) [reply web/src/components/TaskHeader.tsx:25]: I'll remove it from an independent const for now and just put it w the rest of the JSX. Easier to read and no current need for a separate component.

File: web/src/lib/utils/helpers.tsx (hunk 7 above)
[the remark this answers is not in the document]
Comment (aloverso) [reply web/src/lib/utils/helpers.tsx:93]: Ah I see your point

File: web/src/lib/utils/markdownReader.ts (hunk 4 above)
Comment (aloverso) [reply web/src/lib/utils/markdownReader.ts:66]: oops. the way github rendered the diff made me misread this one. all good
